presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/DatabaseShardManager.java

```diff
@@ -719,6 +719,7 @@ public static String shardIndexTable(long tableId)
 
     public static String minColumn(long columnId)
     {
+        checkArgument(columnId >= 0, "invalid columnId %s", columnId);
```

Comment:
Not sure, but all the "special ones" start at -1 and this check is consistent with that. 0 just happens to be unused, but it can be a valid column id.